api-server/web-server/src/main.rs

```diff
@@ -47,17 +56,43 @@ async fn main() {
         &chain_config,
     )
     .await
-    .unwrap_or_else(|e| {
-        log::error!("Error creating Postgres storage: {}", e);
-        std::process::exit(1);
-    });
+    .map_err(ApiServerWebServerInitError::PostgresConnectionError)?;
+
+    let rpc_auth = match (args.rpc_cookie_file, args.rpc_username, args.rpc_password) {
+        (None, None, None) => {
+            let cookie_file_path =
+                default_data_dir_for_chain(chain_type.name()).join(COOKIE_FILENAME);
+            RpcAuthData::Cookie { cookie_file_path }
+        }
+        (Some(cookie_file_path), None, None) => RpcAuthData::Cookie {
+            cookie_file_path: cookie_file_path.into(),
+        },
+        (None, Some(username), Some(password)) => RpcAuthData::Basic { username, password },
+        _ => {
+            return Err(ApiServerWebServerInitError::InvalidConfig(
+                "Invalid RPC cookie/username/password combination".to_owned(),
+            ))
+        }
+    };
```

Comment:
The user shouldn't have to go through the hassle of authentication if submission is disabled.